File: apps/codemap-search/tests/fixtures/extract/sample.java
/** Fixture exercising Java branch-sensitive extraction. */
public class Sample {

    /** Fetch the current value. */
    public String fetch() {
        String label = "fetched value";
        return label;
    }

    /** Old fetcher. */
    @Deprecated
    public String legacyFetch() {
        return "legacy";
    }

    String packagePrivateHelper() {
        return "package-private";
    }

    @Test
    public void testFetch() {
        new Sample().fetch();
    }
}

File: apps/codemap-search/tests/fixtures/extract/sample.kt
/** Fixture exercising Kotlin branch-sensitive extraction. */
class Repository {

    /** Fetch a record. */
    fun fetch(): String {
        val query = "select * from users"
        return query
    }

    /** Old fetcher. */
    @Deprecated("use fetch instead")
    fun legacyFetch(): String {
        return "legacy"
    }

    private fun privateHelper(): Int {
        return 0
    }

    @Test
    fun testFetch() {
        fetch()
    }
}

/** An exported top-level function. */
fun publicHelper(): Int {
    return 1
}
